### ISSUE TITLE
Warnings and exceptions polluting the log

### ISSUE THREAD

**Issue by eXistenZNL** [SiCKRAGETV/sickrage-issues#404]
Branch -> Master
Commit hash -> 33e95870e0c1b241c93f541d4ee27ed6ba3bfcb8
Your operating system and python version -> Debian 7.8, Python 2.7.3
What you did -> Using SickRage via Nginx remote proxy
What happened -> Lots of log messages
What you expected -> No errors :)

Hi,

since recently I'm running SickRage and I love it thus far. The only problem I have is that the logs are polluted with exceptions regarding a failed webui callback. In the web interface, I can see this log:

```python
2015-01-13 17:32:14 Thread-62 :: Failed doing webui callback: Traceback (most recent call last):
  File "/opt/sickbeard/sickbeard/webserve.py", line 230, in async_call
    result = function(**kwargs)
  File "/opt/sickbeard/sickbeard/webserve.py", line 374, in showPoster
    return self.redirect(static_image_path)
  File "/opt/sickbeard/sickbeard/webserve.py", line 191, in redirect
    super(BaseHandler, self).redirect(url, permanent, status)
  File "/opt/sickbeard/tornado/web.py", line 630, in redirect
    self.finish()
  File "/opt/sickbeard/tornado/web.py", line 892, in finish
    self.flush(include_footers=True)
  File "/opt/sickbeard/tornado/web.py", line 846, in flush
    start_line, self._headers, chunk, callback=callback)
  File "/opt/sickbeard/tornado/http1connection.py", line 385, in write_headers
    self._pending_write = self.stream.write(data)
  File "/opt/sickbeard/tornado/iostream.py", line 348, in write
    self._handle_write()
  File "/opt/sickbeard/tornado/iostream.py", line 800, in _handle_write
    _merge_prefix(self._write_buffer, num_bytes)
  File "/opt/sickbeard/tornado/iostream.py", line 1382, in _merge_prefix
    if len(deque) == 1 and len(deque[0]) <= size:
TypeError: object of type 'NoneType' has no len()
```

The corresponding message in the log file on disk is this:

```python
2015-01-13 17:32:14 ERROR    Uncaught exception GET /packages/sickbeard/showPoster/?show=276564&which=poster_thumb (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:8081', method='GET', uri='/packages/sickbeard/showPoster/?show=276564&which=poster_thumb', version='HTTP/1.0', remote_ip='127.0.0.1', headers={'Accept-Language': 'en-us', 'Accept-Encoding': 'gzip, deflate', 'X-Forwarded-For': 'xxx.xxx.xxx.xxx', 'Connection': 'close', 'Accept': 'image/png,image/*;q=0.8,*/*;q=0.5', 'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:34.0) Gecko/20100101 Firefox/34.0 FirePHP/0.7.4', 'Host': 'localhost:8081', 'X-Forwarded-Proto': 'https', 'Referer': 'https://xxx.xxx.xxx.xxx/packages/sickbeard/home/displayShow?show=276564', 'X-Insight': 'activate', 'X-Real-Ip': 'xxx.xxx.xxx.xxx', 'Cookie': 'template=phpsysinfo; _pk_id.5.xxx=xxxxxxxxxxxxxxxxxxxxxxxxxxxxx8.', 'Authorization': 'Basic xxxxxxxxxxxxxxxxxxxxxxxxxxx'})
Traceback (most recent call last):
  File "/opt/sickbeard/tornado/web.py", line 1292, in _stack_context_handle_exception
    raise_exc_info((type, value, traceback))
  File "/opt/sickbeard/tornado/stack_context.py", line 314, in wrapped
    ret = fn(*args, **kwargs)
  File "/opt/sickbeard/tornado/concurrent.py", line 226, in <lambda>
    lambda future: callback(future.result()))
  File "/opt/sickbeard/lib/concurrent/futures/_base.py", line 400, in result
    return self.__get_result()
  File "/opt/sickbeard/lib/concurrent/futures/_base.py", line 359, in __get_result
    reraise(self._exception, self._traceback)
  File "/opt/sickbeard/lib/concurrent/futures/_compat.py", line 107, in reraise
    exec('raise exc_type, exc_value, traceback', {}, locals_)
  File "/opt/sickbeard/lib/concurrent/futures/thread.py", line 61, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/sickbeard/sickbeard/webserve.py", line 230, in async_call
    result = function(**kwargs)
  File "/opt/sickbeard/sickbeard/webserve.py", line 374, in showPoster
    return self.redirect(static_image_path)
  File "/opt/sickbeard/sickbeard/webserve.py", line 191, in redirect
    super(BaseHandler, self).redirect(url, permanent, status)
  File "/opt/sickbeard/tornado/web.py", line 630, in redirect
    self.finish()
  File "/opt/sickbeard/tornado/web.py", line 892, in finish
    self.flush(include_footers=True)
  File "/opt/sickbeard/tornado/web.py", line 846, in flush
    start_line, self._headers, chunk, callback=callback)
  File "/opt/sickbeard/tornado/http1connection.py", line 385, in write_headers
    self._pending_write = self.stream.write(data)
  File "/opt/sickbeard/tornado/iostream.py", line 348, in write
    self._handle_write()
  File "/opt/sickbeard/tornado/iostream.py", line 800, in _handle_write
    _merge_prefix(self._write_buffer, num_bytes)
  File "/opt/sickbeard/tornado/iostream.py", line 1382, in _merge_prefix
    if len(deque) == 1 and len(deque[0]) <= size:
TypeError: object of type 'NoneType' has no len()
```

I tested the develop branch as well and it exhibited the same behaviour (amongst other problems I encountered so I switched back to master).

The thing is, it only seems to happen when using Nginx as a reverse proxy to localhost:8081 on which SickRage is listening. When using SickRage directly on port 8081 the errors seem to stop appearing.

I might have misunderstood the various options in SickRage regarding listening IP, docroot and proxy headers that could prevent this if set up correctly, but i'm unsure if any decent documentation on this topic exists...

If more info is needed, please let me know!

**Comment by garvinob** [comment]
@eXistenZNL Same issue #266 - Waiting a fix, dev has been away for a bit it seems. It only happens behind nginx proxy, though I havent tried any other proxy in front of SR

**Comment by eXistenZNL** [comment]
@garvinob thanks for the heads up!

**Comment by markheloking** [comment]
@echel0n has just had his child born: he's taking some personal time off.

**Comment by abeloin** [comment]
This is fixed with PR https://github.com/SiCKRAGETV/SickRage/pull/1419

**Comment by abeloin** [comment]
PR merged in develop

**Comment by fernandog** [comment]
@eXistenZNL please update and report back

**Comment by eXistenZNL** [comment]
Hi,

already looked at the merge diff when the 'A new version is available" bar displayed and and saw this had possibly been fixed. Great job! after a bit of testing I did last weekend it seems that the errors are indeed gone. I am currently unable to reach my NAS so I am unable to do proper testing, but will do some final testing by clicking through the gui like a mad man for a few minutes once I can reach my NAs again, testing like that should generate a warning fairly quickly if it's not fixed properly.

Lately I looked into the bug some more on my own and I it seemed like images are being redirected to a different URL upon clearing the browser's cache and reloading the page, this seems to be causing the warnings. I might be wrong there and it doesn't really matter as long as it's fixed now ^^

Anyway, will report back with the final results.
